src/tiler/healthCheck.js

```diff
@@ -0,0 +1,105 @@
+var Step = require('windshaft/node_modules/step');
+var Pg = require('windshaft/node_modules/cartodb-psql/node_modules/pg');
+var Redis = require('windshaft/node_modules/redis-mpool/node_modules/redis');
+
+/**
+ * Attempts to connect to a PostgreSQL database, providing the
+ * callback with a success or failure boolean.
+ * @param {string} host
+ * @param {string} port
+ * @param {string} user
+ * @param {string} password
+ * @param {string} database
+ * @param {function} callback
+ */
+function checkPostgres(host, port, user, password, database, callback) {
+  var conString = ['postgres://', user, ':', password, '@', host, '/', database].join('');
+  var client = new Pg.Client(conString);
+
+  client.connect(function(err) {
+    client.end();
+    callback(null, !err);
+  });
+}
+
+/**
+ * Attempts to connect to a Redis data store, providing the
+ * callback with a success or failure boolean.
+ * @param {string} host
+ * @param {string} port
+ * @param {function} callback
+ */
+function checkRedis(host, port, callback) {
+  var client = Redis.createClient(port, host);
+  var wrapper = function(status) {
+    client.quit();
+    callback(null, status);
+  };
+
+  client.on('connect', function() {
+    wrapper(true);
+  });
+
+  client.on('error', function() {
+    wrapper(false);
+  });
+}
+
+/**
+ * Makes use of the PostgreSQL and Redis checks above to return
+ * a JSON object representing the health of both data stores:
+ *
+ * {
+ *   "databases": [
+ *     {
+ *       "default": {
+ *         "ok": true
+ *       }
+ *    }
+ *   ],
+ *   "caches": [
+ *     {
+ *       "default": {
+ *         "ok": true
+ *       }
+ *     }
+ *   ]
+ * }
+ *
+ * @param {object} config - The configuration object passed to Windshaft.Server()
+ */
+module.exports = function createHealthCheckHandler(config) {
+  var pgConf = config.grainstore.datasource,
+
+    pgHost = pgConf.host,
+    pgPort = pgConf.port,
+    pgUser = pgConf.user,
+    pgPassword = pgConf.password,
+    pgDatabase = pgConf.dbname,
```

Comment:
How did you get it to fail this way? I feel like I tested this with the database up and down, but never got this failure.

Should be fixed in https://github.com/WikiWatershed/model-my-watershed/commit/c048f8d6de5e8209ebcb9d2019503d62203c8a02.